### ISSUE TITLE
MDsveX compatible?

### ISSUE THREAD

**Issue by janosh** [atomiks/rehype-pretty-code#73]
Has anyone tried using `rehype-pretty-code` with MDsveX? Just installing it and adding it to the list of `rehypePlugins` doesn't seem to work.

Steps I tried (without any apparent effect)

```sh
pnpm add -D rehype-pretty-code shiki
```

followed by

```js
import adapter from '@sveltejs/adapter-static'
import { mdsvex } from 'mdsvex'
import rehype_pretty_code from 'rehype-pretty-code'
import preprocess from 'svelte-preprocess'

/** @type {import('@sveltejs/kit').Config} */
export default {
  extensions: [`.svelte`, `.svx`, `.md`],

  preprocess: [
    preprocess(),
    mdsvex({
      rehypePlugins: [rehype_pretty_code], // doesn't appear to work
      extensions: [`.svx`, `.md`],
    }),
  ],

  kit: {
    adapter: adapter(),
  },
}
```

Related issue: https://github.com/pngwn/MDsveX/issues/388

**Comment by atomiks** [comment]
This is a standard Rehype plugin, so I'm not sure why it wouldn't work. Maybe the fact that it is asynchronous and isn't being awaited? Either way, it seems like it would be an MDsveX issue.

**Comment by SBV2020** [comment]
@janosh , I have managed to get it working. The thing is, you need to disable svelte highlight. I have post [my config here](https://github.com/rockchalkwushock/rehype-code-titles/issues/255#issuecomment-1815711075) so you could give it a try.

**Comment by SBV2020** [comment]
But @atomiks, I think there is some issue with the rehype-pretty-code. Some of the symbol is being encoded while when using shiki, it doesn't seem to be a problem. 

Using Shiki config: 
![image](https://github.com/atomiks/rehype-pretty-code/assets/144010580/5ea2b3b5-2226-4b46-820f-2cb2ee01e62b)

Using Rehype Pretty Code:
![image](https://github.com/atomiks/rehype-pretty-code/assets/144010580/b323faff-aea8-4a14-961f-cc1995c2d8a6)


I used the Shiki config from [this blog](https://joyofcode.xyz/sveltekit-markdown-blog#syntax-highlighting)
```
const mdsvexOptions = {
	extensions: ['.md'],
	highlight: {
		highlighter: async (code, lang = 'text') => {
			const highlighter = await shiki.getHighlighter({ theme: 'poimandres' })
			const html = escapeSvelte(highlighter.codeToHtml(code, { lang }))
			return `{@html \`${html}\` }`
		}
	},
}
```

**Comment by brunobely** [comment]
Also seeing encoded symbols, any workarounds?

**Comment by xyven1** [comment]
I also ran into the encoding issue with the symbols. After some digging it appears that it has to do with how MDSvex handles the symbols in order to deal with symbols which are used in svelte source code. I also found an [alternative plugin](https://github.com/johnhooks/highlighter) which avoids the issue and uses shiki, so maybe by looking at how that plugin handles the special symbols could help with creating a patch for rehype-pretty-code. The alternative plugin has many similar features to this plugin, but is missing things like inline highlighting and context aware inline highlighting.
